### PR TITLE
[Term Entry] Adding Pytorch Tensor zeros

### PR DESCRIPTION
**Pull request by selenahuang017**
Description
Adding new entry on the .zeros() term under PyTorch Tensors concept. It is under docs/content/pytorch/concepts/tensors/terms/zeros/zeros.md.

**NOTE**: I have included a link to the Tensors concept page (or where it will go theoretically) but I know that the Tensor concept page isn't published yet, so the link currently will not work.

Issue Solved
Addresses issue: https://github.com/Codecademy/docs/issues/4937

Type of Change
Adding a new entry

Checklist
[x ] All writings are my own.
[x ] My entry follows the Codecademy Docs style guide.
[x ] My changes generate no new warnings.
[x ] I have performed a self-review of my own writing and code.
[x ] I have checked my entry and corrected any misspellings.
[x ] I have made corresponding changes to the documentation if needed.
[x ] I have confirmed my changes are not being pushed from my forked main branch.
[x ] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
[x ] I have linked any issues that are relevant to this PR in the Issues Solved section.

### CODE REVIEW

**Comment by selenahuang017** [comment]
Sorry also I can't figure out how to fix the prettier format issue in the check, and I can't tell what the differences are from the github logs. I installed prettier locally but it says Cannot find module '@codecademy/prettier-config'

Any tips would be appreciated. Thank you!

File: content/pytorch/concepts/tensors/terms/zeros/zeros.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+Tags:
+  - 'AI'
+  - 'Data Structures'
+  - 'Arrays'
+  - 'Deep Learning'
+CatalogContent:
+  - 'intro-to-py-torch-and-neural-networks'
+  - 'py-torch-for-classification'
+  - 'paths/machine-learning'
+---
```
Comment (mamtawardhani) [inline content/pytorch/concepts/tensors/terms/zeros/zeros.md:17]:
```suggestion
---
Title: '.zeros()'
Description: 'Initializes a new tensor filled with zeros, with a specified shape.'
Subjects:
  - 'AI'
  - 'Data Science'
  - 'Machine Learning'
Tags:
  - 'AI'
  - 'Arrays'
  - 'Data Structures'
  - 'Deep Learning'
CatalogContent:
  - 'intro-to-py-torch-and-neural-networks'
  - 'py-torch-for-classification'
  - 'paths/data-science'
---
```

Comment (mamtawardhani) [reply content/pytorch/concepts/tensors/terms/zeros/zeros.md:17]: Modifications:
1. title is `.zeros`
2. description should start with verbs
3. Tags and subjects should be in alphabetical order

File: content/pytorch/concepts/tensors/terms/zeros/zeros.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+
+The `.zeros()` method returns a [Tensor](https://www.codecademy.com/resources/docs/pytorch/tensors) of a specified shape with zeros. Optionally, the output tensor, datatype, layout, device, and whether autograd records operations can be specified.
```
Comment (mamtawardhani) [inline content/pytorch/concepts/tensors/terms/zeros/zeros.md:20]:
```suggestion
The **`.zeros()`** method returns a tensor of a specified shape filled with zeros. Optionally, the output tensor, its datatype, layout, device, and whether autograd records operations can be selected.
```

Comment (mamtawardhani) [reply content/pytorch/concepts/tensors/terms/zeros/zeros.md:20]: Modifications: 
1. the first instance of a term should be in bold and backticks
2. There is no entry of tensors as of now, so removed the link

File: content/pytorch/concepts/tensors/terms/zeros/zeros.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+## Syntax
+
+```py
```
Comment (mamtawardhani) [inline content/pytorch/concepts/tensors/terms/zeros/zeros.md:24]:
```suggestion
```pseudo
```

Comment (mamtawardhani) [reply content/pytorch/concepts/tensors/terms/zeros/zeros.md:24]: Syntax is written using the pseudo block

File: content/pytorch/concepts/tensors/terms/zeros/zeros.md [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+import torch
+
+torch.zeros(shape)
+
+# with defaults explicitly stated
```
Comment (mamtawardhani) [inline content/pytorch/concepts/tensors/terms/zeros/zeros.md:29]:
```suggestion
```

Comment (mamtawardhani) [reply content/pytorch/concepts/tensors/terms/zeros/zeros.md:29]: Can be removed

File: content/pytorch/concepts/tensors/terms/zeros/zeros.md [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+torch.zeros(shape, out=None, dtype=torch.float32, layout=torch.strided, device=torch.device('cpu'), requires_grad=False)
```
Comment (mamtawardhani) [inline content/pytorch/concepts/tensors/terms/zeros/zeros.md:30]:
```suggestion
torch.zeros(size, out=None, dtype=None, layout=torch.strided, device=None, requires_grad=False)
```

Comment (mamtawardhani) [reply content/pytorch/concepts/tensors/terms/zeros/zeros.md:30]: Reefrence: https://pytorch.org/docs/stable/generated/torch.zeros.html

File: content/pytorch/concepts/tensors/terms/zeros/zeros.md [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+```
+
+The shape is specified as a variable, tuple, or list of integers. Optional parameters are as follows:
+
+- `out`: The output Tensor, defaults to None.
+- `dtype`: Datatype (`torch.dtype`) of the zeros, defaults to the global default.
+- `layout`: Layout (`torch.layout`) of the output Tensor, defaults to `torch.strided`.
+- `device`: Device (`torch.device`) of the output Tensor, defaults to the global default.
+- `requires_grad`: Boolean of whether autograd will record operations on the output Tensor, defaults to False.
```
Comment (mamtawardhani) [inline content/pytorch/concepts/tensors/terms/zeros/zeros.md:39]: Please refer to the link: https://pytorch.org/docs/stable/generated/torch.zeros.html

only the `layout` parameter has a default value other than global, all others have `None` as their default. Please reframe the parameter descriptions. Also add a bullet point for the `size` parameter

File: content/pytorch/concepts/tensors/terms/zeros/zeros.md [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+
+The returned tensors are as follows:
+
+```shell
+tensor([ 0.,  0., 0.])
+
+tensor([[ 0,  0],
+        [ 0,  0],
+        [ 0,  0],
+        [ 0,  0]])
```
Comment (mamtawardhani) [inline content/pytorch/concepts/tensors/terms/zeros/zeros.md:61]:
```suggestion
tensor([0., 0., 0.])
tensor([[0, 0],
        [0, 0],
        [0, 0],
        [0, 0]], dtype=torch.int16)
```

File: content/pytorch/concepts/tensors/terms/zeros/zeros.md [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+- `device`: Device (`torch.device`) of the output Tensor, defaults to the global default.
+- `requires_grad`: Boolean of whether autograd will record operations on the output Tensor, defaults to False.
+
+## Example
+
+```python
+import torch
+
+t0 = torch.zeros(3)
+t1 = torch.zeros((4, 2), dtype=torch.int16)
```
Comment (mamtawardhani) [inline content/pytorch/concepts/tensors/terms/zeros/zeros.md:47]: Could you add comments here on what is happening in these two lines?

**Review by mamtawardhani — CHANGES_REQUESTED** [review]
Thank you for your contribution to Codecademy Docs @selenahuang017, the entry is nicely written! 😃 

I've suggested a few changes, could you review and modify those at your earliest convenience? Feel free to dicuss any concerns, thank you! 😄

File: content/pytorch/concepts/tensors/terms/zeros/zeros.md (hunk 5 above)
Comment (selenahuang017) [reply content/pytorch/concepts/tensors/terms/zeros/zeros.md:30]: I see. I saw that in the description it said that if the dtype is written as None, the global default will be assigned, which, if not assigned, is torch.float32. Should we still leave it as none?

**Comment by selenahuang017** [comment]
Hi! Not sure why the pull_request is failing, but I checked the logs and there was nothing there to explain what exactly the difference was, especially since the lint passed.

**Comment by mamtawardhani** [comment]
Hey @selenahuang017, the entry looks fine now, however, an example codebyte is also needed under the ##Codebyte Example block, could you please add a codebyte example also?
Thank you! 😃

**Review by mamtawardhani — APPROVED** [review]
Thank you for your contribution @selenahuang017, the entry looks good for a second round of review! 🚀

File: content/pytorch/concepts/tensors/terms/zeros/zeros.md [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+  - 'Machine Learning'
+Tags:
+  - 'AI'
+  - 'Arrays'
+  - 'Data Structures'
+  - 'Deep Learning'
+CatalogContent:
+  - 'intro-to-py-torch-and-neural-networks'
+  - 'py-torch-for-classification'
+  - 'paths/data-science'
```
Comment (SaviDahegaonkar) [inline content/pytorch/concepts/tensors/terms/zeros/zeros.md:16]:
``` suggestion
CatalogContent:
  - 'intro-to-py-torch-and-neural-networks'
  - 'py-torch-for-classification'
  - 'paths/data-science'
```
Only use the course/path that is already present in the [catalog-content. md file ](documentation/catalog-content.md)and else if you do not have the paths that you want to include in the PR present in the list then add one.

File: content/pytorch/concepts/tensors/terms/zeros/zeros.md [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+---
+
+The **`.zeros()`** method returns a tensor of a specified shape filled with zeros. Optionally, the output tensor, its datatype, layout, device, and whether autograd records operations can be selected.
```
Comment (SaviDahegaonkar) [inline content/pytorch/concepts/tensors/terms/zeros/zeros.md:19]:
```suggestion
The **`.zeros()`** method returns a tensor of a specified shape filled with zeros. Optionally, the output tensor, its datatype, layout, device, and whether autograd records operations can be selected.
```
You can rewrite this sentence for more readability.

File: content/pytorch/concepts/tensors/terms/zeros/zeros.md [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+import torch
+
+# Define a tensor with three values in one row
+t0 = torch.zeros(3)
+
+# Define a tensor with four rows, two values per row. All values have the datatype torch.int16
+t1 = torch.zeros((4, 2), dtype=torch.int16)
+
+print(t0)
+print(t1)
```
Comment (SaviDahegaonkar) [inline content/pytorch/concepts/tensors/terms/zeros/zeros.md:48]:
```suggestion
import torch

# Define a tensor with three values in one row
t0 = torch.zeros(3)

# Define a tensor with four rows, two values per row. All values have the datatype torch.int16
t1 = torch.zeros((4, 2), dtype=torch.int16)

print(t0)
print(t1)
```

**Review by SaviDahegaonkar — CHANGES_REQUESTED** [review]
Hey @selenahuang017,
I went through your file and found some changes that should be made once to get your PR merged. Please go through them thoroughly and implement the changes asap.

Thanks,
Savi

File: content/pytorch/concepts/tensors/terms/zeros/zeros.md (hunk 9 above)
Comment (selenahuang017) [reply content/pytorch/concepts/tensors/terms/zeros/zeros.md:16]: sorry i don't see the file, is there another link i can see what courses there are?

File: content/pytorch/concepts/tensors/terms/zeros/zeros.md (hunk 11 above)
Comment (selenahuang017) [reply content/pytorch/concepts/tensors/terms/zeros/zeros.md:48]: I am unclear as to what has changed here?

**Comment by selenahuang017** [comment]
sorry I am not clear on what exactly some of the changes requested are, please let me know if I am missing anything! I will change the one explanation for clarity

File: content/pytorch/concepts/tensors/terms/zeros/zeros.md (hunk 9 above)
Comment (selenahuang017) [reply content/pytorch/concepts/tensors/terms/zeros/zeros.md:16]: oh i found it sorry nvm

**Comment by mamtawardhani** [comment]
> sorry I am not clear on what exactly some of the changes requested are, please let me know if I am missing anything! I will change the one explanation for clarity

@SaviDahegaonkar could you check this on priority? Thank you.

**Comment by SaviDahegaonkar** [comment]
Hey @selenahuang017,
I am sorry for the inconvenience from my side especially for the code part your code is correct, There was a typo error from my side. Is there something else that I can help you with? Please let me know

**Comment by mamtawardhani** [comment]
> Hey @selenahuang017, I am sorry for the inconvenience from my side especially for the code part your code is correct, There was a typo error from my side. Is there something else that I can help you with? Please let me know

@SaviDahegaonkar Please check the recent changes and if all the changes are met with your requirements, please merge this PR on priority since it has been over a month.
Thank you! 😃

**Review by SaviDahegaonkar — APPROVED** [review]
Hey @selenahuang017,
All changes are updates so I hereby approve it.

Thanks,
Savi

**Pull request merged**